modelforge/potential/parameters.py

```diff
@@ -81,6 +81,30 @@ class ActivationFunctionParamsEnum(CaseInsensitiveEnum):
     ELU = ActivationFunctionParamsAlpha
 
 
+class PredictedPropertiesMixin:
```

Comment:
I don't think this should be a mixin here, as it isn't really necessary and I think just adds some complexity. 

The way I was handling these was to just create a separate pydantic class, and just set this as the type for the given parameter (e.g., see how ActivationFunction is set up). 

So we could just define

`class PredictedProperties(BaseModel):
    predicted_properties: List[str]
    predicted_dim: List[int]

    @model_validator(mode="after")
    def check_predicted_properties(self) -> PredictedProperties:
        if len(self.predicted_properties) != len(self.predicted_dim):
            raise ValueError(
                "The length of predicted_properties and predicted_dim must be the same"
            )

        return self`

And then in the each model, under CoreParameter, just add in `predicted: PredictedProperties`, e.g., like:


`class ANI2xParameters(ParametersBase):
    class CoreParameter(ParametersBase):
        angle_sections: int
        maximum_interaction_radius: Union[str, unit.Quantity]
        minimum_interaction_radius: Union[str, unit.Quantity]
        number_of_radial_basis_functions: int
        maximum_interaction_radius_for_angular_features: Union[str, unit.Quantity]
        minimum_interaction_radius_for_angular_features: Union[str, unit.Quantity]
        angular_dist_divisions: int
        activation_function_parameter: ActivationFunctionConfig
        predicted: PredictedProperties
`

The specific reason I did it this way was it keeps the model definitions reasonably self contained  which will make documentation easier.  We might also want to make the properties that are allowed to be predicted an enum, but we can do that later.

Reply:
As pointed out in my last comment, I don't think this is a good idea. But I am happy to take any suggestions!